src/components/PopupWithForm.js

```diff
@@ -8,6 +8,7 @@ export default class PopupWithForm extends Popup {
     this._inputList = [
       ...this._popupForm.querySelectorAll(this._inputSelector),
     ];
+    this._submitButton = this._popupForm.querySelector(".modal__save-button");
     this._handleFormSubmit = handleFormSubmit;
   }
 
```

Comment:
**Could be improved**
You can make a special method for getting a form element from `PopupWithForm` not to duplicate the searching in `index.js`



```js

  getForm() {
    return this._popupForm
  }


```

Reply:
I understand how this could generally be more efficient, but I'm not sure I understand what duplicate searching you're referring to in index.js

Reply:
> I understand how this could generally be more efficient, but I'm not sure I understand what duplicate searching you're referring to in index.js

Actually, you don't duplicate because you get the froms from `document.forms` (it's a nice way). I didn't see that before I left this comment. 

So, everything is fine